### DISCUSSION TITLE
Cant make ogres server to work

### DISCUSSION THREAD

**Discussion by davaguco** [samcf/ogres#118] [Q&A] [answered]
I installed java and clojure on AWS Amazon linux. Then I cloned the git. I type npm install, then npm start. When I head to http://serveraddress:8080 I just get a blank page with a correct favicon and title but no content.

I opened ports 8080, 5000, 9630.

The commands I used are:

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.5/install.sh | bash
. ~/.nvm/nvm.sh
nvm install --lts

curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
unzip awscliv2.zip
sudo ~/aws/install
sudo ~/aws/install --bin-dir /usr/local/bin --install-dir /usr/local/aws-cli --update
aws configure

(I type the key and secret here)

sudo dnf update
sudo yum install java-17-amazon-corretto-devel

curl -L -O https://github.com/clojure/brew-install/releases/latest/download/linux-install.sh && \
chmod +x linux-install.sh && \
sudo ./linux-install.sh
sudo yum install git

git clone https://github.com/samcf/ogres.git
cd ogres 
npm install 
npm fund 
npm start

**Comment by samcf** [comment, marked as answer]
`npm start` is strictly intended to setup a development environment on your own local machine. I haven't done much work to streamline the setup of a new production server; theres a fair bit of implicit knowledge necessary to do so. You might find this guide helpful https://github.com/samcf/ogres/blob/main/etc/setup-guide.sh

On your own machine, you should be able to run the following to setup both the app client and the WebSocket server:
```sh
npm install;
npm start;
```

In another terminal...
```sh
clojure -X:server/dev
```

The app should be available at `https://localhost:8080`

Hope that helps! Good luck!
